framework/src/main/java/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -1,1388 +1,298 @@
-      AnnotatedTypeFactory typeFactory = wildcard.atypeFactory;
-
-      WildcardType javaWildcardType = wildcard.getUnderlyingType();
-      TypeMirror javaExtendsBound;
-      if (javaWildcardType.getExtendsBound() != null) {
-        javaExtendsBound = javaWildcardType.getExtendsBound();
-      } else {
-        javaExtendsBound = TypesUtils.getObjectTypeMirror(typeFactory.processingEnv);
-      }
-
-      if (wildcard.isTypeArgOfRawType()) {
-        rawTypeWildcards.put(wildcard.getTypeVariable(), wildcard.getUnderlyingType());
-      }
-
-      AnnotatedTypeMirror extendsBound =
-          AnnotatedTypeMirror.createType(javaExtendsBound, typeFactory, false);
-      wildcard.setExtendsBound(extendsBound);
-
-      this.wildcards.put(wildcard.getUnderlyingType(), wildcard);
-
-      TypePathNode extendsNode = currentStructure.addPathNode(new ExtendsBoundNode());
-      visit(extendsBound);
-      currentStructure.removePathNode(extendsNode);
+    private AnnotatedTypeMirror createAnnotatedType(TypeMirror javaType) {
+      return AnnotatedTypeMirror.createType(javaType, atypeFactory, false);
     }
 
     /**
-     * Initialize {@code declaredType}'s type arguments.
+     * Initialize the upper and lower bounds of {@code annotatedTypeVariable}.
      *
-     * @param declaredType declared type whose type arguments are initialized
+     * @param annotatedTypeVariable an annotated type variable
      */
-    private void initializeTypeArgs(AnnotatedDeclaredType declaredType) {
-      DeclaredType underlyingType = (DeclaredType) declaredType.underlyingType;
-      if (underlyingType.getTypeArguments().isEmpty() && !declaredType.isUnderlyingTypeRaw()) {
-        return;
-      }
-      TypeElement typeElement =
-          (TypeElement) declaredType.atypeFactory.types.asElement(underlyingType);
-      List<AnnotatedTypeMirror> typeArgs;
-      if (declaredType.typeArgs == null) {
-        int numTypeParameters = typeElement.getTypeParameters().size();
-        typeArgs = new ArrayList<>(numTypeParameters);
-        for (int i = 0; i < numTypeParameters; i++) {
-          TypeMirror javaTypeArg = getJavaType(declaredType, typeElement.getTypeParameters(), i);
-          AnnotatedTypeMirror atmArg =
-              AnnotatedTypeMirror.createType(javaTypeArg, declaredType.atypeFactory, false);
-          typeArgs.add(atmArg);
-          if (atmArg.getKind() == TypeKind.WILDCARD && declaredType.isUnderlyingTypeRaw()) {
-            ((AnnotatedWildcardType) atmArg).setTypeArgOfRawType();
-          }
-        }
-      } else {
-        typeArgs = declaredType.typeArgs;
+    private void initializeTypeVariable(AnnotatedTypeVariable annotatedTypeVariable) {
+      TypeVariable t = annotatedTypeVariable.getUnderlyingType();
+      if (!annotatedTypeVariable.isDeclaration()) {
+        t = (TypeVariable) TypeAnnotationUtils.unannotatedType(t);
+        typeVariableMap.put(t, annotatedTypeVariable);
```

Comment:
Is method `initializeTypeVariable` only ever called once?  If not, then this seems to be redoing work and overwriting a value in the map.  I have the same question for the method below, `initializeWildcard`.